lib/gcpspanner/get_user_saved_search.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcpspanner
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/spanner"
+)
+
+// UserSavedSearch represents a SavedSearch and a user's attributes related to that SavedSearch.
+type UserSavedSearch struct {
+	SavedSearch
+	// Role will be nil if the user is not authenticated.
+	Role *string `spanner:"Role"`
+	// IsBookmarked will be nil if the user is not authenticated.
+	IsBookmarked *bool `spanner:"IsBookmarked"`
+}
+
+// unauthenticatedUserSavedSearchMapper contains the entityMapper implementation for an unauthenticated user.
+type unauthenticatedUserSavedSearchMapper struct{}
+
+func (m unauthenticatedUserSavedSearchMapper) SelectOne(
```

Comment:
In what usage scenario would we need to get just one public saved query rather than getting all of them?

Reply:
When a user is given a saved search (maybe found on a blog) webstatus.dev?s=ID. It will load up the search for them